kore/src/Kore/Step/Axiom/EvaluationStrategy.hs

```diff
@@ -73,31 +90,64 @@ definitionEvaluation equations =
                 Equation.mapVariables (pure fromVariableName)
                 <$> equations
             term' = TermLike.mapVariables Target.mkUnifiedNonTarget term
-        let -- Attempt an equation, pairing it with its result, if applicable.
-            attemptEquation equation =
-                Equation.attemptEquation condition term' equation
-                >>= return . Bifunctor.second apply
-              where
-                apply = Equation.applyEquation condition equation
-        traverse attemptEquation equations' >>=
-            (partitionEithers >>> \case
-                (_, applied : _) -> do
-                    results <- applied
-                    (return . Applied) AttemptedAxiomResults
-                        { results
-                        , remainders = OrPattern.bottom
-                        }
-                (errors, []) ->
-                    case minError of
-                        Just (Equation.WhileCheckRequires _) ->
-                            (return . NotApplicableUntilConditionChanges)
-                                (SideCondition.toRepresentation condition)
-                        _ -> return NotApplicable
-                  where
-                    minError =
-                        (fmap getMin . getOption)
-                        (foldMap (Option . Just . Min) errors)
-            )
+        result <-
+            attemptEquations
+                (attemptEquationAndAccumulateErrors condition term')
+                equations'
+        case result of
+            Right applicable -> do
+                results <- applicable
+                (return . Applied) AttemptedAxiomResults
+                    { results
+                    , remainders = OrPattern.bottom
+                    }
+            Left minError ->
+                case getMin <$> getOption minError of
+                    Just (Equation.WhileCheckRequires _) ->
+                        (return . NotApplicableUntilConditionChanges)
+                            (SideCondition.toRepresentation condition)
+                    _ -> return NotApplicable
+
+attemptEquationAndAccumulateErrors
+    :: (InternalVariable variable, MonadSimplify simplifier)
+    => SideCondition variable
+    -> TermLike (Target variable)
+    -> Option (Min (AttemptEquationError variable))
+    -> Equation variable
+    -> ExceptRT
+        (simplifier (OrPattern variable))
```

Comment:
This should just be `OrPattern variable`, right? Why is the successful result an action?

Reply:
I'm not sure, it used to be this way as well, and decided to not change it in case it mattered.